sidechain/consensus/aura/src/lib.rs

```diff
@@ -176,12 +189,26 @@ where
 		self.environment.init(header, shard)
 	}
 
+	fn proposing_remaining_duration(&self, slot_info: &SlotInfo<PB>) -> Duration {
+		proposing_remaining_duration(slot_info, duration_now())
+	}
+
 	fn allow_delayed_proposal(&self) -> bool {
 		self.allow_delayed_proposal
 	}
 
-	fn proposing_remaining_duration(&self, slot_info: &SlotInfo<PB>) -> Duration {
-		proposing_remaining_duration(slot_info, duration_now())
+	fn import_latest_parentchain_block(
+		&self,
+		current_latest_imported_header: &PB::Header,
+	) -> Result<PB::Header, ConsensusError> {
+		let maybe_latest_imported_header = self
+			.parentchain_import_trigger
+			.import_all()
+			.map_err(|e| ConsensusError::Other(e.into()))?;
+
+		Ok(maybe_latest_imported_header
+			.map(|b| b.block.header().clone())
+			.unwrap_or_else(|| current_latest_imported_header.clone()))
```

Comment:
Here we import the latest parentchain header. If none was imported (e.g. because the queue was empty), we return the 'current' latest parentchain header.

Reply:
I like how this is not very invasive in the current aura code!